### ISSUE TITLE
Pods not found when using a mixed-case datacenterName

### ISSUE THREAD

**Issue by olim7t** [k8ssandra/k8ssandra-operator#1252]
Given this spec:
```yaml
apiVersion: k8ssandra.io/v1alpha1
kind: K8ssandraCluster
metadata:
  namespace: test
  name: k8s-cluster
spec:
  cassandra:
    serverVersion: 4.1.0
    clusterName: Cql_Cluster
    datacenters:
      - metadata:
          name: k8s-dc1
        datacenterName: Cql_DataCenter1
        size: 1
        racks:
          - name: Cql_Rack1
        storageConfig:
          cassandraDataVolumeClaimSpec:
            storageClassName: standard
            accessModes:
              - ReadWriteOnce
            resources:
              requests:
                storage: 100Mi
```
The cluster errors out with:
```
no pods in READY state found in datacenter k8s-dc1
```

This is because cass-operator [sanitizes the Cassandra pod's DC label](https://github.com/k8ssandra/cass-operator/blob/c273e274310b0b007178ab214b2e32774f3cf819/apis/cassandra/v1beta1/cassandradatacenter_types.go#L573) with `CleanupForKubernetes` (no caps, no underscores):
```
cassandra.datastax.com/datacenter: cql-datacenter1
```

But k8ssandra-operator [uses the unsanitized version](https://github.com/k8ssandra/k8ssandra-operator/blob/9039c0237d2000e9fb8591fd8c77e197aeea46c7/pkg/cassandra/management.go#L137) to build its selector:
```
cassandra.datastax.com/datacenter: Cql_DataCenter1
```

We need to use the same sanitization function in k8ssandra-operator.

We should also consider switching to `CleanLabelValue()` in both operators, because this is what cass-operator uses for the cluster and rack labels. Right now we have inconsistent formats:
```
cassandra.datastax.com/cluster: Cql_Cluster
cassandra.datastax.com/datacenter: cql-datacenter1
cassandra.datastax.com/rack: Cql_Rack1
```

```[tasklist]
### Blocked by
- [ ] https://github.com/k8ssandra/cass-operator/issues/622
```

**Comment by adejanovski** [comment]
I think I poo-pooed here when using the sanitized name for the labels :(
Let's move to `CleanLabelValue()` overall, even if I hate having multiple ways of naming things...

**Comment by olim7t** [comment]
I filed k8ssandra/cass-operator#622.
I'll keep this open until I can re-test with the cass-operator fix.

**Comment by olim7t** [comment]
> I'll keep this open until I can re-test with the cass-operator fix

Actually we still need to sanitize the name in k8ssandra-operator, so this gets its own PR.

**Comment by mk01** [comment]
> > I'll keep this open until I can re-test with the cass-operator fix
> 
> Actually we still need to sanitize the name in k8ssandra-operator, so this gets its own PR.

isn't in this particular case (and the line of the code) simply taking wrong field ? instead of "datacenterName()" it should take "Name" (which complies with k8s rules) and cass-operator uses for the label ?
